### PR TITLE
Gp

### PR DESCRIPTION
**Pull request by davidzqhuang**
Merging in TanimotoGPModel (+ sklearn Gaussian process model integration).

Many many thanks to Pat Walters who introduced us to the model (Tanimoto kernel for GP) and provided the version of Ryan-Rhys Griffith's implementation in https://github.com/PatWalters/yamc which we referenced.

This is truly an amazing model which performs consistently well, although full experiments (on OCE's end) to be conducted.

Ryan-Rhys's original implementation is at https://github.com/Ryan-Rhys/The-Photoswitch-Dataset/blob/master/examples/gp_regression_on_molecules.ipynb.

Some notes: 
- Requires GPFlow + Tensorflow to run this model
- SKLearn version ("GaussianProcessModel") does not perform as well as TanimotoGPModel, even with a defined Tanimoto Kernel. This may be due to either SKLearn internals for processing GP models or errors in calculating gradients by hand
- Saving and loading has been implemented for TanimotoGPModel (tests/test_models.py::test_tanimotogpmodel). The implementation is a bit circuitous to allow for better integration with other potential GPFlow models
- TanimotoGPModel is implemented as a BaseSKLearn model with two estimators TanimotoGPRegressor and TanimotoGPClassifier which are in turn subclasses of GPFlowEstimator. The bulk of the saving and loading code is in GPFlowEstimator which saves the X_train and y_train (required for GPFlow models) as well as a parameter_dict. The parameter_dict is serialized with a separate GPFlow_Parameter class which saves the "value", "name", and "transform". Transform is a tensorflow_probability bijector, and is saved by saving the name of the class and then looking up that class on load. Note, there may be future issues with this and other GPFlow models, but those will be addressed case-by-case.

### CODE REVIEW

**Comment by davidzqhuang** [comment]
Three other changes:

- Patches on XGBoostModel for saving and loading
- modified parameterize and create_BC to handle dictionaries
- Patches on stacking models to forward arguments for full functionality
